test/FastTests/Issues/RavenDB-20292.cs

```diff
@@ -56,7 +56,9 @@ await using (var insert = store.BulkInsert())
                 }
                 sw.Stop();
 
+#if MEM_GUARD_STACK
```

Comment:
That looks suspicious

Reply:
It is not, it is not possible to ensure that you are going to be finishing below 5 seconds when you are doing all this validation work. There is no reason to test this under validation; but it does make sense to run the rest of the test for validating memory accesses.

Reply:
You are adding the check *when you are doing this*
did you meant `!MEM_GUARD_STACK` ?

Reply:
That's correct. 🤦‍♂️ also added extra information for why to clarify the rationale.